queue: Improve null-pointer safety

tsaiiuo opened the pull request:
### Summary

This PR enhances the robustness and consistency of the queue module by refining null-pointer safety and improving comment clarity.

### Changes

- Aligned the safety behavior of `include/lib/queue.h` with the pattern already established in `include/lib/list.h` for better module consistency.
- Added null-pointer checks in `include/lib/queue.h` to prevent undefined behavior when accessing invalid queue pointers.
- Prior to these changes, the `queue_peek()` and `queue_dequeue()`  in `lib/queue.c` did not check whether the `q` pointer was null before dereferencing it. Null checks have now been added.
- Removed redundant null checks in `lib/queue.c`, consolidating safety logic where appropriate.
- Added an explicit null check in `queue_enqueue()` to allow the function to fail gracefully with an error code when passed a null pointer.
- Made minor comment improvements throughout the queue module to fix formatting and clarify parameter documentation.

### Benefits

- Prevents crashes due to unchecked null access
- Promotes consistent null-handling practices across modules
- Improves overall readability and maintainability of the queue codebase

jserv commented on include/lib/queue.h:
```diff
@@ -46,19 +46,21 @@ int32_t queue_destroy(queue_t *q);
 /* Returns the number of elements currently in the queue. */
 static inline uint32_t queue_count(const queue_t *q)
 {
+    if (!q)
+        return 0u;
```
Add `unlikely` macro.

jserv commented on include/lib/queue.h:
```diff
@@ -46,19 +47,21 @@ int32_t queue_destroy(queue_t *q);
 /* Returns the number of elements currently in the queue. */
 static inline uint32_t queue_count(const queue_t *q)
 {
+    if (unlikely(!q))
+        return 0u;
```
Unless the whole queue API formally treats `NULL` as a valid (empty) queue, remove the early-return and enforce non-`NULL` via validation at a higher layer. If the `NULL == empty` model is desirable, state it in the header comment and make every queue routine tolerate NULL; otherwise we will inherit new undefined-behavior sites later.

tsaiiuo replied:
To ensure consistency, the updated queue API now tolerates NULL across all functions except `queue_count()`. Treating NULL as an empty queue in `queue_count()` aligns with this design and avoids requiring extra checks before calls like `queue_is_empty()`, which already handles NULL.

Leaving `queue_count()` unchanged could lead to undefined behavior in cases like `mo_mq_destroy()` in `kernel/mqueue.c`, where the parameter is not explicitly checked for being a null pointer.

jserv commented on include/lib/queue.h:
```diff
@@ -11,6 +11,7 @@
 #pragma once
 
 #include <lib/libc.h>
+#include <private/utils.h>
```
Use `#include "private/utils.h"` when it is necessary.

tsaiiuo commented:
@jserv  Hi, I've updated the note in `queue_count()` to clearly state that a NULL queue pointer is treated as an empty queue. I've also rebased and squashed the related commits to keep the history clean.

visitorckw commented:
Thanks for the improvements on safety and documentation.

However, I have some concerns about changing the behavior from "undefined behavior with clear failure" (i.e., kernel panic due to NULL pointer dereference) to "graceful handling of NULL pointers."

Previously, if a developer mistakenly passed a NULL queue pointer, it would result in a crash that immediately revealed the bug and its location. With the new behavior, such mistakes may silently fail or return error codes, potentially making it harder to trace the root cause during development.

Do we really want to make passing NULL pointers a valid use case, rather than treating it as a programming error?
